src/components/Fund/FundDetails/FundDetailsContainer.js

```diff
@@ -56,67 +58,73 @@ const FundDetailsContainer = ({
   resources,
   stripes,
 }) => {
-  useEffect(() => {
-    mutator.fund.reset();
-    mutator.fundType.reset();
-    mutator.ledger.reset();
-    mutator.allocatedFrom.reset();
-    mutator.allocatedTo.reset();
-    mutator.group.reset();
-    mutator.budgets.reset();
-
-    mutator.fund.GET().then(response => {
-      const {
-        fund: {
-          ledgerId,
-          fundTypeId,
-          allocatedFromIds,
-          allocatedToIds,
-        },
-      } = response;
+  const fetchFundDetails = useCallback(
+    () => {
+      mutator.fund.reset();
+      mutator.fundType.reset();
+      mutator.ledger.reset();
+      mutator.allocatedFrom.reset();
```

Comment:
do you really need to re-fetch all the data? IMHO, some of it could be loaded once on didMount

Reply:
fixed

Reply:
but nothing was changed...

Reply:
I've added one more `useEffect` for fetching fund resource only if tags are updated

Reply:
and how it resolves my initial request?)

Reply:
When `params.id` is changed component fetches all resources, and when only tags are changed component re-fetches only fund resource. Maybe I missed smth.